### DISCUSSION TITLE
Secure Connection Between Sec Onion and Winlogbeats

### DISCUSSION THREAD

**Discussion by ColeVan** [Security-Onion-Solutions/securityonion#2410] [Unsupported Versions]
In order to setup a secure connection between Security Onion and our Endpoints, would we modify the elasticsearch.yml and place the necessary certs in the directories the elasticsearch.yml points to?   I'm assuming we uncomment the xpack security line along with the line below? Afterwards do we forward these certs to our endpoints as well? Still looking for documentation that clearly outlines how to setup secure connections between winlogbeats and Sec Onion. Anyone have any advice or writeups on how to accomplish this?
![image](https://user-images.githubusercontent.com/70167373/102831817-ab309d00-43ba-11eb-8e21-bfc2018eecaf.png)
![image](https://user-images.githubusercontent.com/70167373/102831838-bd124000-43ba-11eb-82d4-78572cba71e1.png)
![image](https://user-images.githubusercontent.com/70167373/102831900-e763fd80-43ba-11eb-9b43-591f6c44cf67.png)

**Comment by TOoSmOotH** [comment]
Please do not open 9200 to your endpoints. You should send your beats to logstash on the manager.

**Reply by ColeVan** [reply]
Good to know. Is there a reasoning for not opening up port 9200 to endpoints? Just curious...

Also, what is the best way to securely send windows event logs to security onion? I have winlogbeats sending sysmon and windows eventogs to Sec Onion over port 5044 non encrypted

**Reply by ColeVan** [reply]
Perhaps using Wazuh with sysmon would be best?

**Reply by TOoSmOotH** [reply]
Any client with access to 9200 can add, remove, and manipulate your data in Elastic. Also if you have multiple search nodes the data won't get balanced. I am working on some docs right now on how to enable TLS on the default beats port.

**Reply by TOoSmOotH** [reply]
https://docs.securityonion.net/en/latest/beats.html#using-the-beats-default-port-5044-with-encryption

Give that a try.

**Reply by ColeVan** [reply]
So, after following your document I placed all three certs in the directories referenced in the screenshot on top. On the bottom is a screenshot of what my winlogbeat.yml file looks like on the endpoint. Although there is a connection between Sec Onion and my endpoint over port 5044 I can not tell if it's encrypted. I started a wireshark capture and I am unable to tell. 
Any thoughts? Did I place the certs in the wrong directories? Perhaps the endpoints need the certs as well??
![image](https://user-images.githubusercontent.com/70167373/102937210-a1b93a80-4477-11eb-88f3-3efd55cf56f0.png)
![image](https://user-images.githubusercontent.com/70167373/102937419-1b512880-4478-11eb-8438-aee360b29e5d.png)

![image](https://user-images.githubusercontent.com/70167373/102937251-b72e6480-4477-11eb-8fbd-81b2121d633b.png)

**Reply by TOoSmOotH** [reply]
yes you need the certs on the clients as well.
